packages/css/src/types.ts

```diff
@@ -524,10 +536,10 @@ export interface ColorMode {
    */
   accent?: CSS.ColorProperty
 
-  [k: string]: CSS.ColorProperty | ObjectOrArray<CSS.ColorProperty>
+  [k: string]: CSS.ColorProperty | ObjectOrArray<CSS.ColorProperty> | undefined
 }
 
-interface ColorModesScale extends ColorMode {
+type ColorModesScale = ColorMode & {
```

Comment:
`modes` doesn't match ColorMode index signature (and it shouldn't), so we can't use an interface here.

```
Property 'modes' of type '{ [k: string]: ColorMode; } | undefined' is not assignable to string index type 'string | string[] | { [K: string]: string | string[] | ...; } | undefined'.
```

One can still augment `ColorMode`, so we don't lose anything here.